core/main/src/state/openrpc_state.rs

```diff
@@ -64,27 +64,29 @@ impl ProviderRelationSet {
     }
 }
 
+fn is_provider_enabled(method: &str, checksets: &Vec<String>) -> bool {
+    let method_lower_case = method.to_lowercase();
+    for checkset in checksets {
+        if method_lower_case.starts_with(&checkset.to_lowercase()) {
+            return true;
+        }
+    }
+    false
+}
+
 pub fn build_provider_relation_sets(
     openrpc_methods: &Vec<FireboltOpenRpcMethod>,
+    checksets: &Vec<String>,
```

Comment:
Since you've added the new field `provider_registrations: Vec<String>` to `OpenRpcState`, can we please make both `build_provider_relation_sets` and `is_provider_enabled` `OpenRpcState` methods instead of functions (e.g. `build_provider_relation_sets(&self...`), `is_provider_enabled(&self...`) so that we don't have to pass around the provider list?